operator/cmd/syncclusterconfig/sync.go

```diff
@@ -125,3 +135,83 @@ func loadBoostrapYAML(path string) (map[string]any, error) {
 
 	return config, nil
 }
+
+// maybeMergeSuperusers merges the values found in users.txt into our superusers
+// list found in our bootstrap.yaml. This only occurs if an entry for superusers
+// actually exists in the bootstrap.yaml.
+func maybeMergeSuperusers(logger logr.Logger, clusterConfig map[string]any, path string) {
+	if path == "" {
+		// we have no path to a users.txt, so don't do anything
+		logger.Info("--users-txt not specified. Skipping superusers merge.")
+		return
+	}
+
+	superusersConfig, ok := clusterConfig[superusersEntry]
+	if !ok {
+		// we have no superusers configuration, so don't do anything
+		logger.Info("Configuration does not contain a 'superusers' entry. Skipping superusers merge.")
+		return
+	}
+
+	superusers, err := loadUsersTxt(path)
+	if err != nil {
+		logger.Info(fmt.Sprintf("Error reading users.txt file %q: %v. Skipping superusers merge.", path, err))
+		return
+	}
+
+	superusersAny, ok := superusersConfig.([]any)
+	if !ok {
+		logger.Info(fmt.Sprintf("Unable to cast superusers entry to array. Skipping superusers merge. Type is: %T", superusersConfig))
+		return
+	}
+
+	clusterConfig[superusersEntry] = normalizeSuperusers(append(superusers, mapConvertibleTo[string](superusersAny)...))
+}
+
+func loadUsersTxt(path string) ([]string, error) {
+	usersTxt, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	users := []string{}
+	scanner := bufio.NewScanner(bytes.NewReader(usersTxt))
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		tokens := strings.Split(line, ":")
+		if len(tokens) != 3 {
+			continue
+		}
+		users = append(users, tokens[0])
+	}
+
+	return users, nil
+}
+
+// normalizeSuperusers de-duplicates and sorts the superusers
+func normalizeSuperusers(entries []string) []string {
+	var sorted sort.StringSlice
+
+	unique := make(map[string]struct{})
+	for _, value := range entries {
+		if _, ok := unique[value]; !ok {
+			sorted = append(sorted, value)
+		}
+		unique[value] = struct{}{}
+	}
+
+	sorted.Sort()
+
+	return sorted
+}
+
+func mapConvertibleTo[T any](array []any) []T {
+	converted := []T{}
+	for _, value := range array {
+		if cast, ok := value.(T); ok {
```

Comment:
This comment also applies to the `continue` in `loadUsersTxt`. If we're not going to error out upon unexpected conditions can we include a log line about the values being skipped to leave ourselves a trail of breadcrumbs?

I'd also be fine with panicking or erroring and crashing the job personally but my preference for most things is fail fast and hard.